client/web/compose/src/components/Public/Record/CustomFilterPreset/index.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <div>
+    <b-modal
+      :visible="showModal"
+      :title="$t('recordList.presetFilter.addFilterPresetName')"
+      body-class="p-0"
+      footer-class="flex-column align-items-stretch"
+      centered
+      @hide="onModalHide"
+    >
+      <b-card
+        class="pt-0"
+      >
+        <b-form-group
+          :label="$t('recordList.presetFilter.filterName')"
+        >
+          <b-form-input
+            v-model="filterName"
+            required
+          />
+        </b-form-group>
+      </b-card>
+
+      <template #modal-footer>
+        <div
+          class="d-flex justify-content-between align-items-center"
+        >
+          <b-button
+            variant="light"
+            @click="onReset"
+          >
+            {{ $t('recordList.presetFilter.reset') }}
+          </b-button>
+
+          <div>
+            <b-button
+              variant="link"
+              rounded
+              class="text-primary"
+              @click="onModalHide"
+            >
+              {{ $t('general.label.cancel') }}
+            </b-button>
+            <b-button
+              variant="primary"
+              @click="onSave"
+            >
+              {{ $t('general.label.save') }}
+            </b-button>
+          </div>
+        </div>
+      </template>
+    </b-modal>
+  </div>
+</template>
+
+<script>
+export default {
+  i18nOptions: {
+    namespaces: 'block',
+  },
+
+  name: 'CustomFilterPreset',
+
+  props: {
+    showCustomPresetFilterModal: {
```

Comment:
can this be called `visible`, since based on where it is it is obvious it's the custom filter preset modal (component name)